util/gce/auth.go

```diff
@@ -34,9 +46,104 @@ type AuthToken struct {
 	TokenType string `json:"token_type"`
 }
 
+type AuthTokenProvider interface {
+	GetToken() (string, error)
+	WaitForToken() (string, error)
+}
+
+// AuthTokenProvider is thread-safe.
+type realAuthTokenProvider struct {
+	// Token to use for authentication.
+	token string
+
+	// Protects <token>.
+	mutex sync.RWMutex
```

Comment:
I think it is implementation specific. There is no mutex in fakeAuthTokenProvider.